### PR TITLE
check instance id

### PR DESCRIPTION
**Pull request by piotrek6641**
**What?**  
Added validataion when starting instance with a provided id

**Why?**  ]
When starting sequence there is an option to provide id which was not validated to check if it already exists.
In case of trying to starting multiple instances with same id

**How to verify**
- send sequence, assume it's id is `2b7accb6-1441-4db3-a2f5-f21db28f2be3`
- Start sequence with 
```
curl -X POST \
  '0.0.0.0:8000/api/v1/sequence/2b7accb6-1441-4db3-a2f5-f21db28f2be3/start' \
  --header 'Accept: */*' \
  --header 'Content-Type: application/json' \
  --data-raw '{ "instanceId": "7a0d1492-34b5-4c87-aa09-570a06624f8c" }'
```
  no errors.
- Start sequence with providing ```instanceId``` (existing already)
```
curl -X POST \
  '0.0.0.0:8000/api/v1/sequence/2b7accb6-1441-4db3-a2f5-f21db28f2be3/start' \
  --header 'Accept: */*' \
  --header 'Content-Type: application/json' \
  --data-raw '{ "instanceId": "7a0d1492-34b5-4c87-aa09-570a06624f8c" }'
```
```
409 Conflict
{
  "error": "Instance with a given ID already exists"
}
```
- Start sequence providing invalid ```instanceId```:
```
curl -X POST \
  '0.0.0.0:8000/api/v1/sequence/2b7accb6-1441-4db3-a2f5-f21db28f2be3/start' \
  --header 'Accept: */*' \
  --header 'Content-Type: application/json' \
  --data-raw '{ "instanceId": "NOT_AN_UUID4" }'
```
```
422 Unprocessable Entity
{
  "error": "Invalid Instance id"
}
```

These aspects need to be checked by the reviewer:

- [x] Verify and confirm operation (please post a screenshot) 
- [x] All STH tests pass
- [ ] All [Scramjet Cloud Platform](https://docs.scramjet.org/platform) tests pass
- [ ] Documentation is updated or no changes

### CODE REVIEW

**Comment by piotrek6641** [comment]
#862

**Review by patuwwy — APPROVED** [review]

File: packages/host/src/lib/host.ts [hunk 1]
```diff
@@ -870,6 +871,13 @@ export class Host implements IComponent {
 
         this.logger.info("Start sequence", sequence.id, sequence.config.name);
 
+        if (payload.instanceId && this.instancesStore[payload.instanceId]) {
```
[the remark this answers is not in the document]
Comment (patuwwy) [reply packages/host/src/lib/host.ts:874]: return 422 with proper error message if length doesn't match. also, there should be `verify` method in IDProvider

**Comment by a-tylenda** [comment]
@piotrek6641 when this PR will be ready please mark it with the "ready for review" label. I will run your code and verify if it works as expected. 
Also, it is always nice to have some tips and example commands to run the code and perform fast and smooth verification. You can put them in the PR description. Thanks!

**Review by patuwwy — APPROVED** [review]

File: packages/host/src/lib/host.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (patuwwy) [reply packages/host/src/lib/host.ts]: added a commit with this validation

**Comment by patuwwy** [comment]
> @piotrek6641 when this PR will be ready please mark it with the "ready for review" label. I will run your code and verify if it works as expected. Also, it is always nice to have some tips and example commands to run the code and perform fast and smooth verification. You can put them in the PR description. Thanks!

Desciption updated with test scenarios

**Pull request merged**